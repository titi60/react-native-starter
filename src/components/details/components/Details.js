import React from 'react';

import getTranslations from 'utils/locales';

import Button from 'shared/Button';
import Box from 'shared/Box';
import Text from 'shared/Text';

const Details = ({ navigation }) => (
  <Box flex={1} justifyContent="center" alignItems="center">
    <Text fontSize={4}>{getTranslations('details', 'page_title')}</Text>

    <Button testID="back-button" onPress={() => navigation.goBack(null)}>
      <Text mt={3}>{getTranslations('details', 'navigation_back')}</Text>
    </Button>
  </Box>
);

export default Details;
