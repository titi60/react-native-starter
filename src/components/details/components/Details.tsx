import React, { useCallback, ReactElement } from 'react';

import { DetailsScreenNavigationProp } from '@routes/routes.types';
import getTranslations from '@utils/locales';

import Button from '@shared/Button';
import Box from '@shared/Box';
import Text from '@shared/Text';
import SafeView from '@shared/SafeView';

interface IProps {
  navigation: DetailsScreenNavigationProp;
}

const Details = ({ navigation }: IProps): ReactElement => {
  const goBack = useCallback(() => navigation.goBack(), []);

  return (
    <SafeView>
      <Box flex={1} justifyContent="center" alignItems="center">
        <Text testID="details_title" fontSize={4}>
          {getTranslations('details', 'page_title')}
        </Text>

        <Button testID="back_button" onPress={goBack}>
          <Text mt={3}>{getTranslations('details', 'navigation_back')}</Text>
        </Button>
      </Box>
    </SafeView>
  );
};

export default Details;
