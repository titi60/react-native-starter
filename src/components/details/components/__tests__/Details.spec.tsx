import React from 'react';
import { shallow } from 'enzyme';

import Details from '../Details';

describe('Details page component', () => {
  // given
  const props = { navigation: { goBack: jest.fn() } as any };

  it('should render correctly', () => {
    // When
    const wrapper = shallow(<Details {...props} />);

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger the goBack method when the button is pressed', () => {
    // When
    const wrapper = shallow(<Details {...props} />);

    wrapper
      .findWhere(node => node.prop('testID') === 'back_button')
      .simulate('press');

    // Then
    expect(props.navigation.goBack.mock.calls.length).toBe(1);
  });
});
