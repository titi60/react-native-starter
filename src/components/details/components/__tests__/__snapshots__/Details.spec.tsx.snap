// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page component should render correctly 1`] = `
<SafeView>
  <Styled(Styled(View))
    alignItems="center"
    flex={1}
    justifyContent="center"
  >
    <Styled(Styled(Text))
      color="black"
      testID="details_title"
      variant="xLarge"
    >
      DÃ©tails
    </Styled(Styled(Text))>
    <Button
      onPress={[Function]}
      testID="back_button"
    >
      <Styled(Text)
        color="black"
        mt="medium"
        variant="regular"
      >
        Retour
      </Styled(Text)>
    </Button>
  </Styled(Styled(View))>
</SafeView>
`;
